src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,473 @@
+    {ok, TRef} = timer:send_interval(Interval, status_ping),
+    TRef.
+
+% this function just returns ok, as a return value is communicated back to
+% the gen_server via check_current_queue_length
+-spec check_current_queue_state(pid(), integer()) -> ok.
+check_current_queue_state(ParentPid, DroppedSinceLastCheck) ->
+    case get_max_length() of
+        undefined -> ok;
+                     % max length isn't configured, or something is broken
+                     % don't continue.
+                     %
+        MaxLength ->
+            lager:debug("Queue Monitor max length = ~p", [MaxLength]),
+            check_current_queue_length(ParentPid, MaxLength,
+                                       DroppedSinceLastCheck)
+    end.
+
+% this function just returns ok, as a return value is communicated back to
+% the gen_server via ParentPid ! Message
+-spec check_current_queue_length(pid(), integer(), integer()) -> ok.
+check_current_queue_length(ParentPid, MaxLength, DroppedSinceLastCheck) ->
+    CurrentLength = get_current_length(),
+    case CurrentLength of
+        undefined ->
+            % a queue doesn't appear to be bound to the /analytics
+            % exchange. The only thing we can do is reset the
+            % dropped_since_last_check value to 0
+            ParentPid ! reset_dropped_since_last_check,
+            ok;
+        N ->
+                lager:debug("Queue Monitor current length = ~p", [N]),
+                QueueAtCapacity = CurrentLength == MaxLength,
+                {Ratio, Pcnt} = calc_ratio_and_percent(CurrentLength, MaxLength),
+                case Ratio >= ?LOG_THRESHOLD of
+                    true ->
+                        lager:warning("Queue Monitor has detected RabbitMQ capacity at ~p%", [Pcnt]);
+                    false -> ok
+                end,
+                case QueueAtCapacity of
+                    true ->
+                        lager:warning("Queue Monitor has dropped ~p messages since last check due to queue limit exceeded",
+                                        [DroppedSinceLastCheck]);
+                    false -> ok
+                end,
+                % successfully checked max length and current length
+                % update the state of the gen_server
+                ParentPid ! {MaxLength, N,QueueAtCapacity},
+                ok
+    end.
+
+
+-spec calc_ratio_and_percent(integer(), integer()) -> {float(), float()}.
+calc_ratio_and_percent(0, _MaxLength) ->
+    {0.0, 0.0};
+%% the max_length policy should ensure that CurrentLength <= MaxLength,
+%% but return 100% full if this ever happens
+calc_ratio_and_percent(CurrentLength, MaxLength) when CurrentLength > MaxLength ->
+    {1.0, 100.0};
+calc_ratio_and_percent(CurrentLength, MaxLength) ->
+    Ratio = CurrentLength / MaxLength,
+    Pcnt = round(Ratio * 100.0),
+    {Ratio, Pcnt}.
+
+-spec rabbit_mgmt_server_request(string()) -> oc_httpc:response().
+rabbit_mgmt_server_request(Path) ->
```

Comment:
I think I'd like to see from here on down moved to its own `chef_wm_rabbitmq_management` module, with vhost and queue name being passed in where needed.  Looks like `create_pool` and `delete_pool`  should be in it too. 

This will separate out the stuff that actually talks to the external service into its own interface, which will smooth the way if/when we want to support multiple queues.(And makes it a little cleaner if we need to do an initial direct request from the supervisor before the queue monitor is started.)